engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocationProvider.java

```diff
@@ -211,30 +211,37 @@ public TableLocation getTableLocationIfPresent(@NotNull final TableLocationKey t
         }
     }
 
-    @Override
+    /**
+     * Remove the given table location.
+     *
+     * @apiNote Use with caution: When this is called all subscribers to the specified location will be notified,
+     *          causing them to stop updating. Tables backed by those notifications will end up 'failed'. This location
+     *          provider will continue to update other locations and will no longer provide or request information about
+     *          the deleted location.
+     *
+     * @param locationKey the TableLocation to remove
+     */
     public void removeTableLocationKey(@NotNull final TableLocationKey locationKey) {
         final Object removedLocation = tableLocations.remove(locationKey);
 
+        handleTableLocationKeyRemoved(locationKey.makeImmutable());
         // need to notify subscribers of removed location, and of "removed" size
```

Comment:
This comment should be refined. It now partially applies to the line above, and partially to the line below. Maybe just delete it, it's kind of low value.